pkg/ccl/backupccl/file_sst_sink.go

```diff
@@ -331,6 +331,66 @@ func (s *fileSSTSink) write(ctx context.Context, resp exportedSpan) (roachpb.Key
 	return resp.resumeKey, err
 }
 
+func (s *fileSSTSink) writeKey(
+	ctx context.Context,
+	prefix []byte,
+	key storage.MVCCKey,
+	value storage.MVCCValue,
+	start hlc.Timestamp,
+	end hlc.Timestamp,
+) error {
+	// Initialize the writer if needed.
+	if s.out == nil {
+		if err := s.open(ctx); err != nil {
+			return err
+		}
+	}
+	prefixedKey := key
+	prefixedKey.Key = append(prefix, prefixedKey.Key...)
+	// In the case of compaction, we only keep one flushed file in memory since we are not dealing with
```

Comment:
Yup I think that's pretty straightforward to implement — although I think that only changes when I choose to flush (i.e. don't necessarily flush every time at the end of `processSpanEntry`) and some additional logic to check the keys to make sure they don't overlap/prefixes haven't changed.

But w.r.t. the number of `flushedFiles`, I think it would still only be 1? I'm still only writing one key at a time, so either the key gets added to the `flushedFiles[0]`, or the file sink flushes first before writing the key. We aren't dealing with KV export requests that nicely act as a `BackupManifest_File`. Are there any downsides to having a `BackupManifest_File` that could theoretically span over the entire SST?

Reply:
We actually saw some instability in OR testing when the file entries were larger than 64 MiB. See this [commit.](https://github.com/cockroachdb/cockroach/commit/17c37f24190b927315a19abc06cd87c2a219ba3d) . So, I think we should aim for file entries to be around 64Mib, and ssts to be 128 Mib.